src/treetensornetworks/solvers/tree_sweeping.jl

```diff
@@ -1,76 +1,49 @@
-# 
-# Sweep step
-# 
 
-"""
-  struct SweepStep{V}
-
-Auxiliary object specifying a single local update step in a tree sweeping algorithm.
-"""
-struct SweepStep{V} # TODO: parametrize on position type instead of vertex type
-  pos::Union{Vector{<:V},NamedEdge{V}}
-  time_direction::Int
+function one_site_region(edge; last_edge=false)
+  if last_edge
+    return [src(edge)], [dst(edge)]
+  end
+  return ([src(edge)],)
 end
 
-# field access
-pos(st::SweepStep) = st.pos
-nsite(st::SweepStep) = isa(pos(st), AbstractEdge) ? 0 : length(pos(st))
-time_direction(st::SweepStep) = st.time_direction
-
-# utility
-current_ortho(st::SweepStep) = current_ortho(typeof(pos(st)), st)
-current_ortho(::Type{<:Vector{<:V}}, st::SweepStep{V}) where {V} = first(pos(st)) # not very clean...
-current_ortho(::Type{NamedEdge{V}}, st::SweepStep{V}) where {V} = src(pos(st))
-
-# reverse
-Base.reverse(s::SweepStep{V}) where {V} = SweepStep{V}(reverse(pos(s)), time_direction(s))
-
-function Base.:(==)(s1::SweepStep{V}, s2::SweepStep{V}) where {V}
-  return pos(s1) == pos(s2) && time_direction(s1) == time_direction(s2)
+function two_site_region(edge; last_edge=false)
+  return ([src(edge), dst(edge)],)
 end
 
-# 
-# Pre-defined sweeping paradigms
-# 
+put_kwargs(t::Tuple{Any,NamedTuple}) = t
+put_kwargs(t::Tuple{Any,Any,NamedTuple}) = t
+put_kwargs(t::Tuple{Any,Any,Any,NamedTuple}) = t
 
-function one_site_sweep(
-  direction::Base.ForwardOrdering,
-  graph::AbstractGraph{V},
-  root_vertex::V,
-  reverse_step;
-  kwargs...,
-) where {V}
-  edges = post_order_dfs_edges(graph, root_vertex)
-  steps = SweepStep{V}[]
-  for e in edges
-    push!(steps, SweepStep{V}([src(e)], +1))
-    reverse_step && push!(steps, SweepStep{V}(e, -1))
-  end
-  push!(steps, SweepStep{V}([root_vertex], +1))
-  return steps
-end
-
-function one_site_sweep(direction::Base.ReverseOrdering, args...; kwargs...)
-  return reverse(reverse.(one_site_sweep(Base.Forward, args...; kwargs...)))
-end
+put_kwargs(v::Vector) = (v, (;))
+put_kwargs(j::Integer) = ([j], (;))
+put_kwargs(t::Tuple{<:Integer}) = (t, (;))
+put_kwargs(t::Tuple{<:Integer,<:Integer}) = (t, (;))
```

Comment:
Here's a more elegant and general approach:
```julia
function append_missing_namedtuple(t::Tuple)
  return reverse(prepend_missing_namedtuple(reverse(t)))
end
prepend_missing_namedtuple(t::Tuple) = ((;), t...)
prepend_missing_namedtuple(t::Tuple{<:NamedTuple,Vararg}) = t
```
to circumvent the fact that you can only use a `Vararg` in the last position of an argument list/Tuple.

This gives:
```julia
julia> append_missing_namedtuple((1, 2, 3))
(1, 2, 3, NamedTuple())

julia> append_missing_namedtuple((1, 2, 3, (; x=2)))
(1, 2, 3, (x = 2,))
```

I think this should only accept `Tuple`s of the form `(arg1, arg2, ...)` or `(arg1, arg2, ..., (; kwarg1, kwarg2, ...))`, in the sense of a [FrankenTuple](https://ararslan.github.io/FrankenTuples.jl/latest/), to keep it simpler.

If we need to convert a non-tuple to a tuple, we could have a different pre-processing function `to_tuple` that converts to a tuple if it isn't already:
```julia
to_tuple(x) = (x,)
to_tuple(x::Tuple) = x
```
so then we would use:
```julia
steps = append_missing_namedtuple.(to_tuple.(steps))
```
to canonicalize the steps.